UIMenuItem.cs

```diff
@@ -108,26 +109,26 @@ public virtual void Draw()
 
             if (Hovered && !Selected)
             {
-                _rectangle.Color = Color.FromArgb(20, 255, 255, 255);
+                //_rectangle.Color = Color.FromArgb(20, 255, 255, 255); // Why setting color every time? (I set it in ctor)
                 _rectangle.Draw();
             }
             if (Selected)
                 _selectedSprite.Draw();
 
             _text.Color = Enabled ? Selected ? UnknownColors.Black : UnknownColors.WhiteSmoke : Color.FromArgb(163, 159, 148);
 
-            if (LeftBadge != BadgeStyle.None)
+            if (LeftBadge == BadgeStyle.None)
+            {
```

Comment:
My guess from the change made by the original author (over in the upstream) and after reading the code, is that the user is most likely to create items without badges, this way it will skip an extra condition check most of the times or a fall to the else statement. However, this will probably have the adverse effect for someone using a badge on the left side. (I could be entirely wrong)